From 8060817fcc4fa31c5fc66da988027c4ea5810be9 Mon Sep 17 00:00:00 2001
From: Jonathan Kang <jonathankang@gnome.org>
Date: Wed, 26 Dec 2018 11:18:42 +0800
Subject: [PATCH] Remove pk_is_thread_default check in pk_backend_is_eula_valid

pk_is_thread_default() check was introduced in commit 44bd3ab6 to
detect possible threading problems.

Some backends need call pk_backend_is_eula_valid() in a thread which is
not the default one to perform EULA related actions. This way
pk_is_thread_default() check fails.

Fix that by removing the pk_is_thread_default() check in
pk_backend_is_eula_valid().

Fixes #300
---
 src/pk-backend.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/pk-backend.c b/src/pk-backend.c
index 38f32a04c..2dc464938 100644
--- a/src/pk-backend.c
+++ b/src/pk-backend.c
@@ -955,7 +955,6 @@ pk_backend_is_eula_valid (PkBackend *backend, const gchar *eula_id)
 
 	g_return_val_if_fail (PK_IS_BACKEND (backend), FALSE);
 	g_return_val_if_fail (eula_id != NULL, FALSE);
-	g_return_val_if_fail (pk_is_thread_default (), FALSE);
 
 	present = g_hash_table_lookup (backend->priv->eulas, eula_id);
 	if (present != NULL)
-- 
2.20.1

